docs/NodeDisruptionPolicy.md

```diff
@@ -0,0 +1,146 @@
+      - actions:
+        - type: None
+        path: /var/lib/kubelet/config.json
+      - actions:
+        - reload:
+            serviceName: crio.service
+          type: Reload
+        path: /etc/machine-config-daemon/no-reboot/containers-gpg.pub
+      - actions:
+        - reload:
+            serviceName: crio.service
+          type: Reload
+        path: /etc/containers/policy.json
+      - actions:
+        - type: Special
+        path: /etc/containers/registries.conf
+      sshkey:
+        actions:
+        - type: None
+  readyReplicas: 0
+```
+Say, for instance the user applied the following MachineConfiguration:
+```
+apiVersion: operator.openshift.io/v1
+kind: MachineConfiguration
+metadata:
+  name: cluster
+  namespace: openshift-machine-config-operator
+spec:
+  nodeDisruptionPolicy:
+    files:
+      - path: /etc/my-file
+        actions:
+          - type: None
+```
+Now, the Status will be updated to reflect the merged policy:
+```
+$ oc get MachineConfiguration/cluster -o yaml
+apiVersion: operator.openshift.io/v1
+kind: MachineConfiguration
+metadata:
+  creationTimestamp: "2024-04-16T15:02:37Z"
+  generation: 4
+  name: cluster
+  resourceVersion: "261205"
+  uid: 2c67b155-1898-452f-adbd-ed376afc0ea2
+spec:
+  logLevel: Normal
+  managementState: Managed
+  operatorLogLevel: Normal
+status:
+  nodeDisruptionPolicyStatus:
+    clusterPolicies:
+      files:
+      - actions:
+        - type: None
+        path: /etc/my-file
+      - actions:
+        - type: None
+        path: /etc/mco/internal-registry-pull-secret.json
+      - actions:
+        - type: None
+        path: /var/lib/kubelet/config.json
+      - actions:
+        - reload:
+            serviceName: crio.service
+          type: Reload
+        path: /etc/machine-config-daemon/no-reboot/containers-gpg.pub
+      - actions:
+        - reload:
+            serviceName: crio.service
+          type: Reload
+        path: /etc/containers/policy.json
+      - actions:
+        - type: Special
+        path: /etc/containers/registries.conf
+      sshkey:
+        actions:
+        - type: None
+  readyReplicas: 0
+
+```
+
+
+For this initial implementation the policy supports MachineConfig changes to the following:
+- Files
+- Units
+- sshKeys
+
+The following actions are supported:
+- `None`: No action will be done for this MachineConfig change.
+- `Drain`: This will drain the node of its current workload.
+- `Reload`. Reloads a user specified service. This requires specifying the service to be reloaded in the `reload.serviceName` field.
+- `Restart`. Restarts a user specified service. This requires specifying the service to be restarted in the `restart.serviceName` field
+- `DaemonReload`: This executes a daemon-reload, which reloads the systemd manager configuration.
+- `Reboot`: This will reboot the node.
+- `Special`: This is an internal MCO only action, and cannot be set by the user.
+
+Some key points to note:
```

Comment:
We may want to put it under a separate heading so that these points are better highlighted.

Reply:
Done!